[COH-18373] Enhance/trace levels and context

palikar opened the pull request.

palikar committed:
Implement the filterign of evenet markers in the inspector frontend

palikar committed:
Commit the auto-generated files

palikar committed:
Add metadata visualization for the new events

palikar committed:
Fix the displaying of metadata for some trace events

palikar committed:
Refactor and clean up the new trace filtering frontend; fix some markers' metadat

palikar committed:
Clean up the code and fix the coherent begin/end section

alien1976 reviewed (approved):
Works perfectly!

Suggestions for feature improvements:
1. Multiselect as we discussed in the thread -> https://coherentlabs.slack.com/archives/C7CEWG1CL/p1704357289675799
2. Can we hide the options when `Trace Filtering Enabled` is disabled.
    2.1. Also can we rename the option to `Trace Filtering` if we hide the options.
3. `Trace level` options are not descriptive enough for me but could be for the others. However maybe they will be more descriptive if are named `Level 1`, `Level 2`, `Level 3`
4. Fix the names in the `Trace System` to be `Resource loading` instead of `ResourceLoading`
5. Add `:` to the options so they are consistent with the rest Inspector options -> for example `Trace level:` or `Trace system:`

![image](https://github.com/CoherentLabs/devtools-frontend-fork/assets/29014573/69f9a29c-eb59-46b0-986a-cc2755e4ebb3)

palikar commented:
1. I've created a JIRA [task](https://coherent-labs.atlassian.net/browse/COH-18770) for that.
2. Renamed the field to "Trace Filtering"
3. The levels are now "Level 1/2/3"
4. Fixed the names (this was done on the backend)
5. Added ":"

palikar committed:
Address the review comments; improve the UX a bit (disabling controls when tracing is inactive; text tweaks)

omalinov commented on front_end/generated/InspectorBackendCommands.js:
```diff
@@ -752,6 +752,7 @@ inspectorBackend.registerEnum("Tracing.TracingBackend", {Auto: "auto", Chrome: "
 inspectorBackend.registerEvent("Tracing.bufferUsage", ["percentFull", "eventCount", "value"]);
 inspectorBackend.registerEvent("Tracing.dataCollected", ["value"]);
 inspectorBackend.registerEvent("Tracing.tracingComplete", ["dataLossOccurred", "stream", "traceFormat", "streamCompression"]);
+inspectorBackend.registerCommand("Tracing.getTraceSystemsAndLevels", [], ["systems", "levels"]);
```
```suggestion
// COHERENT BEGIN
inspectorBackend.registerCommand("Tracing.getTraceSystemsAndLevels", [], ["systems", "levels"]);
// COHERENT END
```

palikar replied:
this is a generated file. All `COHERENT BEGIN` / `COHERENT END` comments for changes responsible for this are in `browser_protocol.pdl`

omalinov commented on front_end/generated/protocol-mapping.d.ts:
```diff
@@ -3546,6 +3546,10 @@ export namespace ProtocolMapping {
       paramsType: [Protocol.Tethering.UnbindRequest];
       returnType: void;
     };
+    'Tracing.getTraceSystemsAndLevels': {
+      paramsType: [];
+      returnType: Protocol.Tracing.GetTraceSystemsAndLevelsResponse;
+    };
```
```suggestion
    // COHERENT BEGIN
    'Tracing.getTraceSystemsAndLevels': {
      paramsType: [];
      returnType: Protocol.Tracing.GetTraceSystemsAndLevelsResponse;
    };
    // COHERENT END
```

palikar replied:
this is a generated file. All `COHERENT BEGIN` / `COHERENT END` comments for changes responsible for this are in `browser_protocol.pdl`

omalinov commented on front_end/generated/protocol-proxy-api.d.ts:
```diff
@@ -3053,6 +3053,8 @@ declare namespace ProtocolProxyApi {
   }
 
   export interface TracingApi {
+    invoke_getTraceSystemsAndLevels(): Promise<Protocol.Tracing.GetTraceSystemsAndLevelsResponse>;
```
```suggestion
    // COHERENT BEGIN
    invoke_getTraceSystemsAndLevels(): Promise<Protocol.Tracing.GetTraceSystemsAndLevelsResponse>;
    // COHERENT END
```

palikar replied:
this is a generated file. All `COHERENT BEGIN` / `COHERENT END` comments for changes responsible for this are in `browser_protocol.pdl`

omalinov commented on front_end/generated/protocol.d.ts:
```diff
@@ -12993,6 +12993,19 @@ declare namespace Protocol {
       System = 'system',
     }
 
+    /**
+     * COHERENT_BEGIN
+     */
+    export interface TraceCategoryDefinition {
+      name: string;
+      categoryString: string;
+    }
+
+    export interface GetTraceSystemsAndLevelsResponse extends ProtocolResponseWithError {
+      systems: TraceCategoryDefinition[];
+      levels: TraceCategoryDefinition[];
+    }
```
```suggestion
    // COHERENT BEGIN
    export interface TraceCategoryDefinition {
      name: string;
      categoryString: string;
    }

    export interface GetTraceSystemsAndLevelsResponse extends ProtocolResponseWithError {
      systems: TraceCategoryDefinition[];
      levels: TraceCategoryDefinition[];
    }
    // COHERENT END
```

palikar replied:
this is a generated file. All `COHERENT BEGIN` / `COHERENT END` comments for changes responsible for this are in `browser_protocol.pdl`

omalinov commented on front_end/panels/timeline/TimelinePanel.ts:
```diff
@@ -420,6 +460,55 @@ export class TimelinePanel extends UI.Panel.Panel implements Client, TimelineMod
+      this._traceSystemsDefinitions = res.systems;
+      this._traceLevelsDefinitions = res.levels;
+
+      this._controller?.dispose();
+      this._controller = null;
```
This should break the world when you receive the response while tracing - e.g. when you refresh and trace a page (not sure if it occurs, but seems possible.

What if instead we get the tracing agent directly, as we seem to construct the controller for it.
Maybe through those calls https://github.com/CoherentLabs/devtools-frontend-fork/blob/13706efede840af5ccdc8ed8d62b38d4d310ab63/front_end/core/protocol_client/InspectorBackend.ts#L733

The following seems okay at a first glance, do you find issues with it?
```JS
fetchTraceSystemsAndLevels(agent) {
    const response = await agent.invoke_getTraceSystemsAndLevels();
    return response;
}

fetchTraceSystemsAndLevels(tracingAgent()).then(res => {
  // setup stuff
});
```

palikar replied:
sure, I reworked the tracing agent acquisition logic based on the suggestion and it seems fine. I've tested the new logic and everything works correctly (refreshing during tracing leads to no crashes) .

palikar committed:
Address the review comments
